### ISSUE TITLE
The 'python3-xlib==0.*,>=0.15.0' distribution was not found and is required by streamdeck-ui

### ISSUE THREAD

**Issue by dhtseany** [timothycrosley/streamdeck-ui#212]
**Describe the bug**
I'm the new streamdeck-ui package maintainer for Arch Linux and I'm working on upgrading us from 1.1.3 to 2.0.3. I'm catching a problem where the app is complaining that python3-xlib wasn't of sufficient version value even thought I presently have python-xlib-0.31-3, which I believe to be of sufficient version compatibility. 

**To Reproduce**
Launch _streamdeck_ from the terminal on a current Arch Linux system. You'll need an updated PKGBUILD to truly install and test it, which I can provide upon request. 

**Expected behavior**
The app to launch without error.

**Screenshots**
![python-xlib-problem](https://user-images.githubusercontent.com/24855489/157523844-45821865-146d-4519-a12a-0f59bbe707e5.png)

**Operating system (please complete the following information)**
 - OS: Arch Linux [Current to 07MAR2022]
 - Desktop environment: Gnome 3
 - Python version: Latest, Python 3.10

**Stream Deck Version**
 - XL

**Comment by dodgyrabbit** [comment]
Hi @dhtseany - thanks for reporting this issue. I had a look and it seems like the root cause is that streamdeck_ui has a dependency on python3-xlib, which is considered a different package from python-xlib.

There are two options:
1. You include python3-xlib
2. I change streamdeck_ui to use python-xlib (my preference)

It seems to me that python3-xlib is abandoned and was originally created around 2014 as an attempt to support new language features. Since then, Python 3+ support has been added to python-xlib. My feeling is that the correct solution is to change the dependency to python-xlib. I did some very basic testing and it seems OK to me. FYI This library is required because of pynput.

I'll have to publish a new build with the updated dependency.

**Comment by dhtseany** [comment]
Thank you for your helpful reply, it's much appreciated! 

I've been looking around Arch's repos, both AUR and the regular package repos, to see what options I have for sourcing python3-xlib and while we still have python2-xlib it seems, as you've also seen, that the python3-xlib upstream was left behind and the Arch maintainers stopped supporting it in favor of python-xlib. 

How much effort do you think it'd take to update the dependency to python-xlib? I'm pretty sure most package maintainers will be cool with the update since most of the other major distros all seem to have python-xlib available now.

**Comment by AceofCookies** [comment]
Anything us poor Arch users can do to work around this while a fix is in the works?

**Comment by Cinux90** [comment]
@AceofCookies 
I have the same problem as you, additionally no deep knowledge regarding the Python structure in the OS, therefore no guarantee that it works error free.
But with the following commands I was able to start the software.

```
cp -r /usr/lib/python3.10/site-packages/python_xlib-0.31-py3.10.egg-info python3_xlib-0.31-py3.10.egg-info
sed -i 's/Name: python-xlib/Name: python3-xlib/' /usr/lib/python3.10/site-packages/python3_xlib-0.31-py3.10.egg-info/PKG-INF
```

**Comment by Cinux90** [comment]
@dhtseany 
Not sure if this can be part of the PKGBUILD as post task I'm not familiar with the PKGBUILD. But this would be a workaround for now.

**Comment by stepfister** [comment]
Hi @Cinux90 , 

I'm using the actual linux manjaro, updated for 2 days. I have the same problem with "python3-xlib". I tried your workaround, but it doesn't work. But I don't really understand what do you do with your workaround.

1.) with the cp -r line you make a directory branch in your home folder?
2.) with the sed line you make what? Do you change one line in /usr/lib/python3.10/site-packages/python3_xlib-0.31-py3.10.egg-info/PKG-INF ?

If I paste your lines in my terminal: The sed command didn't work. Error: Directory not found. 

Maybe you can explain your fix a little bit more in detail for a noob like me?

Thanks, 
Stefan

**Comment by stepfister** [comment]
@dodgyrabbit How can I include python3-xlib on Linux Manjaro in the meantime until the new build comes out with the fix? Thanks for your good work.

**Comment by Cinux90** [comment]
@stepfister 
Indeed there is a typo
```
# copy python_xlib directory in into the same parent folder with different name (python3_xlib...)
cp -r /usr/lib/python3.10/site-packages/python_xlib-0.31-py3.10.egg-info /usr/lib/python3.10/site-packages/python3_xlib-0.31-py3.10.egg-info
# change the, i hope so, package name from python-xlib to python3-xlib
sed -i 's/Name: python-xlib/Name: python3-xlib/' /usr/lib/python3.10/site-packages/python3_xlib-0.31-py3.10.egg-info/PKG-INF
```
If you do that as normal user you need to add a `sudo` in front of the cp and sed command

PS: I updated my initial comment too.

**Comment by dodgyrabbit** [comment]
I've committed the change to remove the python3-xlib dependency.  However, I need to test this with a couple of clean installs to make sure it works properly. I also want to test some upgrade scenarios. Will keep you posted.

**Comment by dodgyrabbit** [comment]
I published version 2.0.4 on test.pypi.org to try it out.

For Ubuntu I did this:
```
sudo apt install python3-pip libhidapi-libusb0 libxcb-xinerama0
```
For arch there should be something similar specifically to install pip and libhidapi. The libxcb-xinerama0 is a workaround for an issue with pyside. It may not be applicable on arch.

From a clean install you have to first manually install the dependencies:
```
python3 -m pip install filetype==1.0.10 cairosvg==2.5.2 streamdeck==0.9.1 pillow==9.0.0 pynput==1.7 pyside2==5.15 python-xlib==0.31
```
Next install 2.0.4
```
python3 -m pip install -i https://test.pypi.org/simple/ streamdeck-ui==2.0.4 --user
```
You need to add the udev rules as per the main instructions.

I tested on a clean Ubuntu 2004 instance and everything worked as expected.
If you're having trouble with the install from test.pypi.org - I may just publish to the main repo. It seems like a fairly safe change.

**Comment by dodgyrabbit** [comment]
Version 2.0.4 has been published which does not depend on python3-xlib. Let me know if you have any success now to create the streamdeck-ui package on Arch.
